docs/how-to/c-functions.rst

```diff
@@ -0,0 +1,375 @@
+
+Before we can call the function, we need to look it up first. To do this, we need to find and load the library in which the function is defined. In the case of standard C functions, this is the standard C library, ``libc``. Because this library is commonly used, Rubicon already loads it by default and exposes it in Python as :attr:`rubicon.objc.runtime.libc`.
+
+.. code-block:: python
+
+    >>> from rubicon.objc.runtime import libc
+    >>> libc
+    <CDLL '/usr/lib/libc.dylib', handle 7fff60d0cb90 at 0x105850b38>
+
+.. note::
+
+    For a list of all C libraries that Rubicon loads and exposes by default, see the :ref:`predefined-c-libraries` section of the :mod:`rubicon.objc.runtime` reference documentation.
+    
+    To access a library that is not predefined by Rubicon, you can use the :func:`~rubicon.objc.runtime.load_library` function:
+    
+    .. code-block:: python
+    
+        >>> from rubicon.objc.runtime import load_library
+        >>> libm = load_library("m")
+        >>> libm
+        <CDLL '/usr/lib/libm.dylib', handle 7fff60d0cb90 at 0x10596be10>
+
+C functions are accessed as attributes on their library:
+
+.. code-block:: python
+
+    >>> libc.puts
+    <_FuncPtr object at 0x110178f20>
+
+However, unlike Objective-C methods, we cannot call C functions right away --- we must first declare the function's argument and return types. (Rubicon cannot do this automatically like with Objective-C methods, because plain C doesn't provide the runtime type information necessary for this.) This type information is found in C header files, in this case ``stdio.h`` (which defines standard C input/output functions, including ``puts``).
+
+The exact location of the macOS C headers varies depending on your version of macOS and the developer tools --- it is not a fixed path. To open the header directory in the Finder, run the following command in the terminal:
+
+.. code-block:: sh
+
+    $ open "$(xcrun --show-sdk-path)/usr/include"
+
+.. note::
+
+    This command requires a version of the macOS developer tools to be installed. If you do not have Xcode or the command-line developer tools installed yet, run this command in the terminal to install the command-line developer tools:
+
+    .. code-block:: sh
+
+        $ xcode-select --install
+
+Once you have opened the relevant header file in a text editor, you need to search for the declaration of the function you're looking for. In the case of ``puts``, it looks like this:
+
+.. code-block:: c
+
+    int puts(const char *);
+
+This means that ``puts`` returns an ``int`` and takes a single argument of type ``const char *`` (a C string). This translates to the following Python ``ctypes`` code:
+
+.. code-block:: python
+
+    >>> from ctypes import c_char_p, c_int
+    >>> libc.puts.restype = c_int
+    >>> libc.puts.argtypes = [c_char_p]
+
+Now that we have provided all of the necessary type information, we can call ``libc.puts``:
+
+.. code-block:: python
+
+    >>> libc.puts(b"Hello!")
```

Comment:
The fact that a C string is a Python byte string is worth a brief explanation. We probably don't want to get into the weeds of explaining how to do Unicode in C, other than "it's hard, turn back now before you go mad...". :-)